docs/docs/javascript/overview/eslint.mdx

```diff
@@ -0,0 +1,554 @@
+config_discovery = true
+
+# Skip linting and formatting entirely
+skip = false
+
+# Enable JSON output with summary parsing
+json_output = true
+
+# Additional TypeScript/plugin dependencies
+extra_dev_dependencies = [
+    "@typescript-eslint/parser@^6.0.0",
+    "@typescript-eslint/eslint-plugin@^6.0.0"
+]
+```
+
+### Version and Requirements
+
+By default, Pants uses ESLint version 8.57.0. You can customize this:
+
+```toml title="pants.toml"
+[eslint]
+version = "eslint@9.0.0"
+```
+
+## Usage
+
+### Linting
+
+Run ESLint to check for code quality issues:
+
+```bash
+$ pants lint src/js/::
+```
+
+### Formatting
+
+Run ESLint with `--fix` to automatically fix formatting issues:
+
+```bash
+$ pants fmt src/js/::
+```
+
+## BUILD File Configuration
+
+### JavaScript Sources
+
+For JavaScript files, use the `js_sources` target:
+
+```python title="src/js/BUILD"
+js_sources(
+    name="frontend",
+    sources=["**/*.js", "**/*.mjs", "**/*.cjs"],
+)
+```
+
+### TypeScript Sources
+
+For TypeScript files, use the `typescript_sources` target:
+
+```python title="src/ts/BUILD"
+typescript_sources(
+    name="backend",
+    sources=["**/*.ts"],
+)
+```
+
+
+### Mixed Projects
+
+For projects with multiple file types:
+
+```python title="src/BUILD"
+js_sources(
+    name="js_files",
+    sources=["**/*.js"],
+)
+
+typescript_sources(
+    name="ts_files",
+    sources=["**/*.ts"],
+)
+```
+
+### Target-specific Configuration
+
+You can skip ESLint for specific targets using the `skip_eslint` field:
+
+```python title="src/js/BUILD"
+js_sources(
+    name="legacy",
+    sources=["legacy/*.js"],
+    skip_eslint=True,  # Skip linting for legacy code
+)
+
+typescript_sources(
+    name="modern",
+    sources=["modern/*.ts"],
+    # ESLint will run on these files
+)
+```
+
+## Configuration Files
+
+### Flat Config (ESLint 9.0+)
+
+```javascript title="eslint.config.js"
+export default {
+  languageOptions: {
+    ecmaVersion: 2022,
+    sourceType: "module",
+  },
+  rules: {
+    "no-unused-vars": "error",
+    "quotes": ["error", "double"],
+    "semi": ["error", "always"],
+  },
+};
+```
+
+### Legacy Config
+
+```json title=".eslintrc.json"
+{
+  "env": {
+    "browser": true,
+    "es2022": true
+  },
+  "extends": ["eslint:recommended"],
+  "rules": {
+    "no-unused-vars": "error",
+    "quotes": ["error", "double"],
+    "semi": ["error", "always"]
+  }
+}
+```
+
+### Package.json Config
+
+```json title="package.json"
+{
+  "name": "my-project",
+  "eslintConfig": {
+    "extends": ["eslint:recommended"],
+    "rules": {
+      "no-unused-vars": "error"
+    }
+  }
+}
+```
+
+## Supported File Types
+
+ESLint in Pants supports:
+
+- JavaScript (`.js`, `.mjs`, `.cjs`)
+- TypeScript (`.ts`) - requires `@typescript-eslint/parser` and `@typescript-eslint/eslint-plugin`
+
+## Integration with Other Tools
+
+### Prettier Integration
+
+ESLint and Prettier can work together. If you're using both tools, consider:
```

Comment:
The numbering implies to me that _all_ of these things should be done in order - I'm not sure that's the case? 

Does Pants support running prettier first, then eslint by default, or we need to consider the order of `pants fmt lint ...` ? 

I do think covering this usage is important FWIW - it's been a bit of a minefield in personal past experience to get these working together. I think if we can offer some opinionated guidance, that would be helpful.